math/src/polynom/multi_linear.rs

```diff
@@ -0,0 +1,240 @@
+use alloc::vec::Vec;
+use core::ops::Index;
+
+use utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
+
+use crate::FieldElement;
+
+// MULTI-LINEAR POLYNOMIAL
+// ================================================================================================
+
+/// Represents a multi-linear polynomial.
+///
+/// The representation stores the evaluations of the polynomial over the boolean hyper-cube
+/// {0 , 1}^ν.
+#[derive(Clone, Debug)]
+pub struct MultiLinearPoly<E: FieldElement> {
+    num_variables: usize,
+    evaluations: Vec<E>,
+}
```

Comment:
Do we actually need to store `num_variables` here? It seems like it can be pretty easily obtained as `self.evaluations.len().ilog2()` or even just `self.evaluations.len().trailing_zeros()`.

Reply:
Actually, it is not needed and we can just call the methods on the vector as you suggested.
Removed